doc/api_assets/service_worker.js

```diff
@@ -0,0 +1,19 @@
+function activateWorkerImmediately(event) {
+  event.waitUntil(self.skipWaiting())
+}
+
+function becomeAvailableToAllPages(event) {
+  event.waitUntil(self.clients.claim())
+}
+
+function cachedResponseOrDoRequest(event) {
+  event.respondWith(
+    caches.open('v1')
+      .then(cache => caches.match(event.request))
+      .then(cachedResponse => cachedResponse || fetch(event.request))
```

Comment:
Went for a trivial caching strategy here, could easily be a lot more fancy if we decided too. See the mentioned cookbook in the PR description for lots of different strategies.